File: ror_record.ts
import { verifyIdentifier } from "./utility.ts";
import { normalizeROR, validateROR } from "./ror.ts";

/**
 * Verifies an ROR using an external database, ror.org
 * @param ror - The identifier string to verify.
 * @returns Promise<boolean> - True if the ROR exists, otherwise false.
 */
export async function verifyROR(ror: string): Promise<boolean> {
  const normalizedROR = normalizeROR(ror);
  return await verifyIdentifier(
    ror,
    normalizedROR,
    validateROR,
  );
}

File: ror_record.go
package metadatatools

// VerifyROR verifies a ROR identifier exists via ror.org. The
// identifier is normalized (and resolves directly to its ror.org URL)
// before verifying.
func VerifyROR(ror string) bool {
	normalized := NormalizeROR(ror)
	return VerifyIdentifier(ror, normalized, ValidateROR)
}
